### PR TITLE
Convert DROP COLUMN SQL to pgroll operation

### PR DESCRIPTION
**Pull request by ryanslade**
Part of https://github.com/xataio/pgroll/issues/504

### CODE REVIEW

**Comment by ryanslade** [comment]
@andrew-farries What should we do about more complicated statements? The fun spec is:

```sql
DROP [ COLUMN ] [ IF EXISTS ] column_name [ RESTRICT | CASCADE ]
```

File: pkg/sql2pgroll/convert.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 	"fmt"
 
 	pgq "github.com/pganalyze/pg_query_go/v6"
+
```
[the remark this answers is not in the document]
Comment (ryanslade) [reply pkg/sql2pgroll/convert.go:9]: I'm using Goland which tries to sort the imports into three groups:

standard library
third party
Imports starting with `github.com/xataio/pgroll`

**Comment by ryanslade** [comment]
@andrew-farries Thanks for the help, this is ready for review.

File: pkg/sql2pgroll/alter_table.go [hunk 2]
```diff
@@ -155,6 +158,32 @@ func convertAlterTableAddUniqueConstraint(stmt *pgq.AlterTableStmt, constraint *
...
+		// Supported
+	case pgq.DropBehavior_DROP_CASCADE:
+		// Fall back to SQL
+		return nil, nil
+	}
+
+	// IF EXISTS not supported
+	if cmd.MissingOk {
+		return nil, nil
+	}
```
Comment (andrew-farries) [inline pkg/sql2pgroll/alter_table.go:178]: For other operations we have the pattern of pulling out these checks into `canConvert` functions:

https://github.com/xataio/pgroll/blob/main/pkg/sql2pgroll/alter_table.go?plain=1#L161-L188

It's up to you if you want to follow the pattern here.

File: pkg/sql2pgroll/alter_table_test.go [hunk 3]
```diff
@@ -72,6 +81,10 @@ func TestUnconvertableAlterTableAddConstraintStatements(t *testing.T) {
 		// operations when changing data type.
 		`ALTER TABLE foo ALTER COLUMN a SET DATA TYPE text COLLATE "en_US"`,
 		"ALTER TABLE foo ALTER COLUMN a SET DATA TYPE text USING 'foo'",
+
+		// CASCADE and IF EXISTS clauses are not represented by OpDropColumn
+		"ALTER TABLE foo DROP COLUMN bar CASCADE",
+		"ALTER TABLE foo DROP COLUMN IF EXISTS bar",
```
Comment (andrew-farries) [inline pkg/sql2pgroll/alter_table_test.go:87]: We should rename the test here now that it's not just covering unconvertible add constraint operations.

Comment (ryanslade) [reply pkg/sql2pgroll/alter_table_test.go:87]: Done

File: pkg/sql2pgroll/alter_table.go (hunk 2 above)
Comment (ryanslade) [reply pkg/sql2pgroll/alter_table.go]: Done

**Comment by ryanslade** [comment]
Once merged, do we mark the entry in the table at https://github.com/xataio/pgroll/issues/504 with orange or green? We still fall back to raw sql in some cases so not sure if that is considered "done done"

**Comment by andrew-farries** [comment]
> Once merged, do we mark the entry in the table at #504 with orange or green? We still fall back to raw sql in some cases so not sure if that is considered "done done"

I think we can set it to 🟢 . The 'fallback to raw sql' testcases serve as a good list of extra options for the respective `pgroll` operations to support, but we don't need to commit to supporting all of them as part of #504.

**Review by andrew-farries — APPROVED** [review]

**Pull request merged**